server/methods/channelsList.js

```diff
@@ -44,19 +44,22 @@ Meteor.methods({
 			}
 		}
 
-		const roomTypes = [];
+		let channels = [];
 
 		if (channelType !== 'private') {
 			if (RocketChat.authz.hasPermission(Meteor.userId(), 'view-c-room')) {
-				roomTypes.push({
-					type: 'c'
-				});
+				if (filter) {
```

Comment:
I'm not sure if this test is right, I think we inverted because we are using filter when we dont have filter 🤔